command/config.go

```diff
@@ -0,0 +1,111 @@
+package command
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	RootCmd.AddCommand(configCmd)
+	configCmd.AddCommand(configGetCmd)
+	configCmd.AddCommand(configSetCmd)
+
+	configGetCmd.Flags().StringP("host", "h", "", "Get per-host setting")
+	configSetCmd.Flags().StringP("host", "h", "", "Set per-host setting")
+
+}
+
+var configCmd = &cobra.Command{
+	Use:   "config",
+	Short: "Set and get gh settings",
+	Long: `Get and set key/value strings. They can be optionally scoped to a particular host.
+
+Current respected settings:
+- git_protocol: https or ssh. Default is https.
+- editor: if unset, defaults to environment variables.
+`,
+}
+
+var configGetCmd = &cobra.Command{
+	Use:   "get",
+	Short: "Prints the value of a given configuration key.",
+	Long: `Get the value for a given configuration key, optionally scoping by hostname.
+
+Examples:
+  gh config get git_protocol
+  https
+  gh config get --host example.com
+  ssh
+`,
+	Args: cobra.ExactArgs(1),
+	RunE: configGet,
+}
+
+var configSetCmd = &cobra.Command{
+	Use:   "set",
+	Short: "Updates configuration with the value of a given key.",
+	Long: `Update the configuration by setting a key to a value, optionally scoping by hostname.
+
+Examples:
+  gh config set editor vim
+  gh config set --host example.com editor eclipse
+`,
+	Args: cobra.ExactArgs(2),
+	RunE: configSet,
+}
+
+func configGet(cmd *cobra.Command, args []string) error {
+	key := args[0]
+	hostname, err := cmd.Flags().GetString("host")
+	if err != nil {
+		return err
+	}
+
+	ctx := contextForCommand(cmd)
+
+	cfg, err := ctx.Config()
+	if err != nil {
+		return err
+	}
+
+	val, err := cfg.Get(hostname, key)
+	if err != nil {
+		return err
+	}
+
+	if val != "" {
+		out := colorableOut(cmd)
+		fmt.Fprintf(out, "%s\n", val)
+	}
+
+	return nil
+}
+
+func configSet(cmd *cobra.Command, args []string) error {
+	key := args[0]
+	value := args[1]
+
+	hostname, err := cmd.Flags().GetString("host")
+	if err != nil {
+		return err
+	}
+
+	ctx := contextForCommand(cmd)
+
+	cfg, err := ctx.Config()
+	if err != nil {
+		return err
+	}
+
+	err = cfg.Set(hostname, key, value)
+	if err != nil {
+		return fmt.Errorf("failed to set %q to %q: %s", key, value, err)
```

Comment:
Nit: to legitimately wrap an error go 1.13-style, could we turn the `%s` into `%w`? I think that the error output will be the same

Reply:
:woman_facepalming: I wasn't sure what the %w was until now and had avoided it out of principle. will switch to it now that I know better.